docs/architecture/architecture.md

```diff
@@ -0,0 +1,671 @@
+- `Length` is the size of the entire message in bytes and encoded in network order.
+- `Payload` is the JSON-encoded command request or response data.
+
+Each stream message is composed of a stream sequence code and a payload containing
+the stream data.
+
+```
+  ┌────────────────┬────────────────┬──────────────────────────────┐
+  │  Sequence Code │     Length     │          Payload             │
+  │   (64 bits)    │    (64 bits)   │     (data length bytes)      │
+  └────────────────┴────────────────┴──────────────────────────────┘
+```
+- `Sequence code` is the 64 bit sequence number assigned by `cc-agent` for a stream.
+- `Length` is the size of the entire stream message in bytes and encoded in network
+  order.
+- `Payload` is the stream data.
+
+The `cc-agent` makes use of [`libcontainer`](https://github.com/opencontainers/runc/tree/master/libcontainer)
+to manage the lifecycle of the container. This way the `cc-agent` reuses most
+of the code used by [`runc`](https://github.com/opencontainers/runc).
+
+## Runtime
+
+`cc-runtime` is an OCI compatible container runtime and is responsible for handling
+all commands specified by
+[the OCI runtime specification](https://github.com/opencontainers/runtime-spec)
+and launching `cc-shim` instances.
+
+`cc-runtime` heavily utilizes the
+[virtcontainers project](https://github.com/containers/virtcontainers), which
+provides a generic, runtime-specification agnostic, hardware-virtualized containers
+library.
+
+### Configuration
+
+The runtime uses a TOML format configuration file called `configuration.toml`. By
+default this file is installed in the `/etc/clear-containers` directory.
+Most users will not need to modify the configuration file.
+
+The file is well commented and provides a few "knobs" that can modify the
+behaviour of the runtime.
+
+The configuration file is also used to enable runtime debug output (see
+https://github.com/clearcontainers/runtime#debugging).
+
+### Significant OCI commands
+
+Here we will describe how `cc-runtime` handles the most important OCI commands.
+
+#### [`create`](https://github.com/clearcontainers/runtime/blob/master/create.go)
+
+When handling the OCI `create` command, `cc-runtime` goes through the following steps:
+
+1. Create the networking container namespace on the host, according to the container
+  OCI configuration file. We only support networking namespaces for now, but
+  will support more of them later.
+2. Run all the [prestart OCI hooks](https://github.com/opencontainers/runtime-spec/blob/master/config.md#hooks)
+  in the host namespaces created in step 1, as described by the OCI container
+  configuration file.
+3. [Set up the container networking namespace up](#networking). This is when
+  all networking interfaces are created and setup inside the namespace, according
+  to the selected pod networking model (CNM for Docker or CNI for Kubernetes).
+4. Create and start the virtual machine running the container process. The VM
```

Comment:
This reads like we start and run the VM and container process(workload) here. I think we actually run the workload process in 'start', so maybe better to:
s/running/that will run/